odoo/modules/loading.py

```diff
@@ -167,15 +167,16 @@ def load_module_graph(
         module_cursor_query_count = env.cr.sql_log_count
         module_extra_query_count = odoo.sql_db.sql_counter
 
-        needs_update = update_module and package.state in ("to install", "to upgrade")
+        needs_update = update_module and (package.state in ("to install", "to upgrade") or module_name in registry._reinit_modules)
```

Comment:
There seem to be three distinct "update" operations on modules: install, upgrade, and reinitialize.  The first one does nothing on installed modules, and the 2nd and 3rd one should do nothing on non-installed modules, right?

What about naming the cases explicitly instead of relying on three different boolean variables?
```suggestion
        update_operation = (
            'install' if package.state == 'to install' else
            'upgrade' if package.state == 'to upgrade' else
            'reinit' if module_name in registry._reinit_modules else
            None
        ) if update_module else None
```
With this, all the `if` statements will look like `if update_operation in (X, Y)`, which might possibly improve readability.  Could you give it a try, and see whether it makes the code more readable?